extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -381,17 +321,15 @@ public long getDislikeCount() throws ParsingException {
     @Override
     public String getUploaderUrl() throws ParsingException {
         assertPageFetched();
-        String uploaderId = null;
         try {
-            uploaderId = getVideoSecondaryInfoRenderer().getObject("owner").getObject("videoOwnerRenderer")
-                    .getObject("navigationEndpoint").getObject("browseEndpoint").getString("browseId");
+            String uploaderUrl = getUrlFromNavigationEndpoint(getVideoSecondaryInfoRenderer()
+                    .getObject("owner").getObject("videoOwnerRenderer").getObject("navigationEndpoint"));
+            if (uploaderUrl != null) return uploaderUrl;
+        } catch (Exception ignored) {}
+        try {
+            String uploaderId = playerResponse.getObject("videoDetails").getString("channelId");
+            if (uploaderId != null) return "https://www.youtube.com/channel/" + uploaderId;
```

Comment:
This should be done using the 
[`YoutubeChannelLinkHandlerFactory.getUrl(iploaderId, null, null)`](https://github.com/TeamNewPipe/NewPipeExtractor/blob/fc465c8bf7b70d9052fc5bf1f524f1c97b330718/extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeChannelLinkHandlerFactory.java#L39-L41) method. But it seems to be buggy to me. It does not return "https://www.youtube.com/channel/ID", but "https://www.youtube.com/ID". I searched for a small random channel (www.youtube.com/user/effectpl) and looked up its channelId (`UCuyC5c8JXzPx71aMg7rGFgQ`). While www.youtube.com/UCuyC5c8JXzPx71aMg7rGFgQ results in a 404 response,  www.youtube.com/channel/UCuyC5c8JXzPx71aMg7rGFgQ is valid.
Does anyone know, why it is implemented that way?

Reply:
That's something I noticed as well. It appears that someone wants to have `channel/UCwhatever` as input there, so that it would also accept e.g. `user/whatever` URLs without having to do a conversion of some kind.

Reply:
Ah yes, correct. That was the reason. Can you please add this info as Jdoc and use the method?